examples/todomvc/src/store.rs

```diff
@@ -0,0 +1,321 @@
+                                            let mut temp_item = Item {
+                                                title,
+                                                completed,
+                                                id,
+                                            };
+                                            item_list.push(temp_item);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        self.data = Some(item_list);
+        &self.data
+    }
+
+    /// Write the local ItemList to localStorage.
+    ///
+    /// todos `ItemList` is the items to store
+    fn set_local_storage(&mut self, todos: ItemList) {
+        let array = js_sys::Array::new();
+        for item in todos.iter() {
+            let mut child = js_sys::Array::new();
+            let s = item.title.clone();
+            child.push(&JsValue::from(&s));
+            child.push(&JsValue::from(item.completed));
+            child.push(&JsValue::from(item.id.to_string()));
+
+            array.push(&JsValue::from(child));
+        }
+        if let Ok(storage_string) = JSON::stringify(&JsValue::from(array)) {
+            let storage_string: String = storage_string.to_string().into();
+            dbg(&storage_string);
+            self.local_storage
+                .set_item(&self.name, storage_string.as_str());
+            self.data = Some(todos);
+        }
+    }
+
+    /// Find items with properties matching those on query.
+    /// `ItemQuery` query Query to match
+    ///
+    /// ```
+    ///  let data = db.find(ItemQuery::Completed {completed: true});
+    ///	 // data will contain items whose completed properties are true
+    /// ```
+    pub fn find(&mut self, query: ItemQuery) -> Option<ItemListSlice> {
+        self.get_local_storage();
+        if let Some(ref todos) = self.data {
+            Some(todos.iter().filter(|todo| query.matches(*todo)).collect())
+        } else {
+            None
+        }
+    }
+
+    /// Update an item in the Store.
+    ///
+    /// `ItemUpdate` update Record with an id and a property to update
+    pub fn update(&mut self, update: ItemUpdate) {
+        let id = update.id();
+        self.get_local_storage();
+        if let Some(todos) = self.data.take() {
+            let todos = todos.into_iter();
+
+            let todos = todos
+                .map(|mut todo| {
+                    if id == todo.id {
+                        todo.update(&update);
+                    }
+                    todo
+                }).collect();
+            self.set_local_storage(todos);
+        }
+    }
+
+    /// Insert an item into the Store.
+    ///
+    /// `Item` item Item to insert
+    pub fn insert(&mut self, item: Item) {
+        self.get_local_storage();
```

Comment:
It might be good to have a precondition to all these methods that they're already fetched from local storage, as then it shouldn't be necessary to load this at the start of all functions I think?